pkg/controller/bundlec/bundle_sync_task.go

```diff
@@ -293,42 +292,40 @@ func (st *bundleSyncTask) updateBundle() error {
 	if err != nil {
 		return errors.Wrap(err, "failed to update bundle")
 	}
-	st.logger.Sugar().Debugf("Set bundle status to %s", &bundleUpdated.Status)
 	st.logger.Sugar().Debugf("Set bundle finalizers to %s", bundleUpdated.Finalizers)
 	return nil
 }
 
+func (st *bundleSyncTask) updateBundleStatus() error {
+	bundleUpdated, err := st.bundleClient.Bundles(st.bundle.Namespace).UpdateStatus(st.bundle)
+	if err != nil {
+		return errors.Wrap(err, "failed to update bundle status")
+	}
+	st.logger.Sugar().Debugf("Set bundle status to %s", &bundleUpdated.Status)
+	return nil
+}
+
 func (st *bundleSyncTask) handleProcessResult(retriable bool, processErr error) (bool /*retriable*/, error) {
 	if processErr != nil && api_errors.IsConflict(errors.Cause(processErr)) {
 		return retriable, processErr
 	}
 	if processErr == context.Canceled || processErr == context.DeadlineExceeded {
 		return false, processErr
 	}
-
-	bundleUpdated := false
-
 	if st.newFinalizers != nil {
 		// Update finalizers
 		st.bundle.Finalizers = st.newFinalizers
-		// Set status to "InProgress: True"
-		// (there will be one more iteration for resource sync that will set an appropriate status)
-		// TODO: Setting status to "InProgress" here might be unnecessary
-		// (if the Bundle spec wasn't updated during the last update, the status should be kept with no changes)
-		// The better solution could be storing "ObservedGeneration" in the status to reflect
-		// whether the status is up-to-date with the spec or stale
-		st.bundle.Status.Conditions = []cond_v1.Condition{
```

Comment:
We are sending an update here to the resource's body (not `/status`) which, when subresource `/status` is enabled, ignores `.status` changes. So there is no point making any changes to `.status` - they are ignored. See https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#status-subresource